web_client/src/components/VtkViewer.vue

```diff
@@ -276,121 +155,258 @@ export default {
+    }
+    function updateCrosshairs() {
+      const myCanvas: HTMLCanvasElement = document.getElementById(`crosshairs-${name.value}`) as HTMLCanvasElement;
       if (myCanvas && myCanvas.getContext) {
         const ctx = myCanvas.getContext('2d');
         ctx.clearRect(0, 0, myCanvas.width, myCanvas.height);
 
-        if (this.showCrosshairs) {
+        if (showCrosshairs.value) {
           const crosshairSet = new CrosshairSet(
-            this.name,
-            this.ijkName,
-            this.representation,
-            this.view,
+            name.value,
+            ijkName.value,
+            representation.value,
+            props.view,
             myCanvas,
-            this.iIndexSlice,
-            this.jIndexSlice,
-            this.kIndexSlice,
+            iIndexSlice.value,
+            jIndexSlice.value,
+            kIndexSlice.value,
           );
           const originalColors = {
             x: '#fdd835',
             y: '#4caf50',
             z: '#b71c1c',
           };
           const trueColors = Object.fromEntries(
-            Object.entries(originalColors).map(([axisName, hex]) => [this.trueAxis(axisName), hex]),
+            Object.entries(originalColors).map(([axisName, hex]) => [trueAxis(axisName), hex]),
           );
           const [displayLine1, displayLine2] = crosshairSet.getCrosshairsForAxis(
-            this.trueAxis(this.name),
+            trueAxis(name.value),
             trueColors,
           );
-          this.drawLine(ctx, displayLine1);
-          this.drawLine(ctx, displayLine2);
+          drawLine(ctx, displayLine1);
+          drawLine(ctx, displayLine2);
         }
       }
-    },
+    }
     /** Place crosshairs at the location of a click event */
-    placeCrosshairs(clickEvent) {
+    function placeCrosshairs(clickEvent) {
       const crosshairSet = new CrosshairSet(
-        this.name,
-        this.ijkName,
-        this.representation,
-        this.view,
+        name.value,
+        ijkName.value,
+        representation.value,
+        props.view,
         null,
-        this.iIndexSlice,
-        this.jIndexSlice,
-        this.kIndexSlice,
+        iIndexSlice.value,
+        jIndexSlice.value,
+        kIndexSlice.value,
       );
       const location = crosshairSet.locationOfClick(clickEvent);
-      this.SET_SLICE_LOCATION(location);
-    },
-    cleanup() {
-      if (this.renderSubscription) {
-        this.renderSubscription.unsubscribe();
-        this.resizeObserver.unobserve(this.$refs.viewer);
+      setSliceLocation(location);
+    }
+    function cleanup() {
+      props.view.setContainer(null);
+      if (renderSubscription.value) {
+        renderSubscription.value.unsubscribe();
+        resizeObserver.value.unobserve(viewer.value);
       }
-      if (this.modifiedSubscription) {
-        this.modifiedSubscription.unsubscribe();
+    }
+    function initializeSlice() {
```

Comment:
Just wanted to confirm removing the if clause that returned if name was default is intentional?

Reply:
Yes, that case never occurs, and this way makes the typing of `name` clearer.